addons/mail/static/src/models/autocomplete-input-view/autocomplete-input-view.js

```diff
@@ -0,0 +1,295 @@
+        },
+        /**
+         * @param {MouseEvent} ev
+         */
+        onBlur(ev) {
+            this._hide();
+        },
+        onFocusin() {
+            this.update({ isFocused: true });
+        },
+        /**
+         * @param {MouseEvent} ev
+         */
+        onKeydown(ev) {
+            if (ev.key === 'Escape') {
+                this._hide();
+            }
+        },
+        /**
+         * @private
+         */
+        _hide() {
+            if (this.discussAsMobileAddItemHeader) {
+                this.discussAsMobileAddItemHeader.clearIsAddingItem();
+            }
+            if (this.discussSidebarCategoryAsAddingItemInputView) {
+                this.discussSidebarCategoryAsAddingItemInputView.onHideAddingItem();
+            }
+            if (this.messagingMenuAsMobileNewMessageInput) {
+                this.messagingMenuAsMobileNewMessageInput.toggleMobileNewMessage();
+            }
+        },
+        /**
+         * @private
+         * @returns {boolean/FieldCommand}
+         */
+        _isFocusOnMount() {
+            if (this.discussAsMobileAddItemHeader) {
+                return true;
+            }
+            if (this.discussSidebarCategoryAsAddingItemInputView) {
+                return true;
+            }
+            if (this.messagingMenuAsMobileNewMessageInput) {
+                return true;
+            }
+            return clear();
+        },
+        /**
+         * @private
+         * @returns {string|FieldCommand}
+         */
+        _computeCustomClass() {
+            if (this.discussSidebarCategoryAsAddingItemInputView) {
+                if (this.discussSidebarCategoryAsAddingItemInputView === this.messaging.discuss.categoryChannel) {
+                    return 'o_DiscussSidebarCategory_newChannelAutocompleteSuggestions';
+                }
+            }
+            if (this.messagingMenuAsMobileNewMessageInput) {
+                return 'o_MessagingMenu_mobileNewMessageInputAutocomplete';
+            }
+        },
+        /**
+         * @private
+         * @returns {boolean|FieldCommand}
+         */
+        _computeIsHtml() {
+            if (this.discussSidebarCategoryAsAddingItemInputView) {
+                return this.discussSidebarCategoryAsAddingItemInputView === this.messaging.discuss.categoryChannel;
+            }
+            return clear();
+        },
+        /**
+         * @private
+         * @returns {string|FieldCommand}
+         */
+        _computePlaceholder() {
+            if (this.chatWindowOwnerAsNewMessageFormInput) {
+                return this.env._t("Search user...");
+            }
+            if (this.discussAsMobileAddItemHeader) {
+                if (this.discussAsMobileAddItemHeader.isAddingChannel) {
+                    return this.discussAsMobileAddItemHeader.addChannelInputPlaceholder;
+                }
+                return this.discussAsMobileAddItemHeader.addChatInputPlaceholder;
+            }
```

Comment:
This is one example why it would be better to split the 2 mobile input immediately, we don't want to handle nested condition every time we have `discussAsMobileAddItemHeader`.